dts/bindings/sensor/st,iis2dlpc-common.yaml

```diff
@@ -33,3 +33,90 @@ properties:
         - 2 # Low Power M3
         - 3 # Low Power M4
         - 4 # High Performance
+
+    pulse-mode:
+      type: int
+      required: false
+      default: 0
+      description: Pulse mode
+      enum:
+        - 0 # Only Single
+        - 1 # Single adn Double
+
+    pulse-thsx:
+      type: int
+      required: false
+      default: 0x0E
+      description: Pulse X-axis threshold
+        (values range from 0x00 to 0x1F)
+        Threshold to start the pulse-event detection procedure on the X-axis.
+        Threshold values for each axis are unsigned 5-bit corresponding
+        to an 2g acceleration full-scale range.
+
+    pulse-thsy:
+      type: int
+      required: false
+      default: 0x0E
+      description: Pulse Y-axis threshold
+        (values range from 0x00 to 0x1F)
+        Threshold to start the pulse-event detection procedure on the Y-axis.
+        Threshold values for each axis are unsigned 5-bit corresponding
+        to an 2g acceleration full-scale range.
+
+    pulse-thsz:
+      type: int
+      required: false
+      default: 0x0E
+      description: Pulse Z-axis threshold
+        (values range from 0x00 to 0x1F)
+        Threshold to start the pulse-event detection procedure on the Z-axis.
+        Threshold values for each axis are unsigned 5-bit corresponding
+        to an 2g acceleration full-scale range.
+
+    pulse-x-enable:
+      type: int
```

Comment:
should this be a boolean instead?

Reply:
Yes, it was my first choice. But then I discovered that booleans don't support the 'default' properties, do they? So, I opted for a normal int, because I want them to default to 1 (true). Does it make sense to you?

Reply:
And if I use boolean withouth default property, then the result is false.

Reply:
No, booleans can't be defaulted.  But if the power-up default is to enable the feature, then perhaps the devicetree should have a boolean property to disable it.

Looking at the data sheet: why is this `pulse`?  The datasheet uses `tap`, which is a more familiar term for accelerometers, isn't it?  And AFAICT tap detection is *not* enabled on power-up.

I'm changing my mind.   What should be there is three integer properties `tap-threshold-x` (`-y`, `-z`), none of which have a default, and presence of a non-zero value means the threshold is enabled on that axis.  No enable properties needed.

The only use of `pulse` seems to be for `DRDY_PULSED` in `CTRL7`.

Reply:
I just maintained the naming that was already in iis2dlpc Kconfig. Which I wrote in the past, so I made that mistake in the past. Let me check better

Reply:
OK.  A case can be made that the renaming is out of scope (though desirable), but in any case the enable state in devicetree should be inferred by a non-zero threshold (unless it's legitimate to enable with a zero threshold, in which case something else has to be done).

And, if we're replacing Kconfig with devicetree, a case can be made that the default behavior should be the same as it was with Kconfig.  In that case the defaults should be documented as retaining the previous defaults from Kconfig, even though they're not the sensor defaults.

I think @galak should weigh in here.

Reply:
> And, if we're replacing Kconfig with devicetree, a case can be made that the default behavior should be the same as it was with Kconfig. In that case the defaults should be documented as retaining the previous defaults from Kconfig, even though they're not the sensor defaults.

Another way to do the same thing would be by adding a `CONFIG_MY_DRIVER_LEGACY_DEFAULTS` variable. If it is `y`, the driver chooses to maintain compatibility with the old behavior when the information isn't in the DT. The option can be removed in a future release after a deprecation period if desirable.

Reply:
@pabigot, I tend to agree. The iis2dlpc driver was taken from lis2dw12, which is a similar ST accelerometer, where the tap and tap-tap feature was introduced here: e4ed17472. So, the naming comes from that commit. But I have to admit that using "tap" in place of "pulse" maybe be much more clear.

Reply:
Sorry guys, I think that I lost few comments here. Let me recap a little bit better. First of all the tap and tap-tap feature has never been tested (at least by me) on iis2dlpc. I had to do it in the plan, but somehow I forgot. So, I guess that having a "legacy" definition is not that important as the feature may not be working already. So, what I'm planning to do, but not immediately unfortunately, is to move everything in dts as I'm doing, rename the names into something like 'tap' and reconsider the using of the three axis enabling properties (instead I may follow the suggested way of considering threshold zero as axis is disabled).

Reply:
As a matter of facts, threshold equal to 0 is not possible, as this setting would result in the device continuosly generating tap events.

Reply:
Is that a useable configuration?  If not, then you should still be able to use absence of the property, or presence with a zero value, as an indicator that the axis should be disabled.

Reply:
> Is that a useable configuration? If not, then you should still be able to use absence of the property, or presence with a zero value, as an indicator that the axis should be disabled.

Exactly. THat is what I was saying. Since it is not a usable configuration I will use zero as indicator to disable the axis.